dotcom-rendering/src/web/components/SignInGate/displayRule.ts

```diff
@@ -66,41 +66,11 @@ export const isValidTag = (tags: TagType[]): boolean => {
 	);
 };
 
-export const hasRequiredConsents = (): Promise<boolean> => {
-	const hasConsentedToAll = (state: ConsentState) => {
-		const consentFlags = state.tcfv2?.consents
-			? Object.values(state.tcfv2.consents)
-			: [];
-		const vendorConsentFlags = state.tcfv2?.vendorConsents
-			? Object.values(state.tcfv2.vendorConsents)
-			: [];
-		const isEmpty =
-			consentFlags.length === 0 || vendorConsentFlags.length === 0;
-
-		return (
-			!isEmpty && [...consentFlags, ...vendorConsentFlags].every(Boolean)
-		);
-	};
-
-	return new Promise((resolve) => {
-		onConsentChange((state) => {
-			if (state.tcfv2) {
-				return resolve(hasConsentedToAll(state));
-			}
-
-			if (state.ccpa) {
-				return resolve(!state.ccpa.doNotSell);
-			}
-
-			if (state.aus) {
-				return resolve(state.aus.personalisedAdvertising);
-			}
-
-			// this shouldn't ever be hit, but this is here as safety
-			return resolve(false);
-		});
-	});
-};
+// check CMP banner consents
+export const hasRequiredConsents = (): Promise<boolean> =>
+	onConsent()
+		.then((consentState: ConsentState) => consentState.canTarget)
```

Comment:
If you love destructuring :)

```suggestion
		.then(({ canTarget }: ConsentState) => canTarget)
```